pkg/envoy/cds.go

```diff
@@ -0,0 +1,134 @@
+package envoy
+
+import (
+	"context"
+	"fmt"
+
+	clusterv3 "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openservicemesh/osm/pkg/utils"
+
+	"github.com/openservicemesh/osm-health/pkg/common"
+)
+
+// HasClusterCheck implements common.Runnable
+type HasClusterCheck struct {
+	ConfigGetter
+
+	dstPod *corev1.Pod
+	k8s    kubernetes.Interface
+}
+
+// Run implements common.Runnable
+func (c HasClusterCheck) Run() error {
+	if c.ConfigGetter == nil {
+		log.Error().Msg("Incorrectly initialized ConfigGetter")
+		return ErrIncorrectlyInitializedConfigGetter
+	}
+	envoyConfig, err := c.ConfigGetter.GetConfig()
+	if err != nil {
+		return err
+	}
+
+	if envoyConfig == nil {
+		return ErrEnvoyConfigEmpty
+	}
+
+	// The destination Pod might back multiple services, so check that at least
+	// one of those services is listed as a cluster in the source Envoy config.
+	possibleClusterNames := map[string]struct{}{}
```

Comment:
I sometimes use `map[string]interface{}` and assign `nil`s

Reply:
One thing I like about `struct{}` is that `struct{}{}` is literally the only value you can assign to a key, vs. `interface{}` where you can assign literally anything. I think using `struct{}` for the value type makes it more clear that the value doesn't matter.